File: src/main/java/com/platformbuilders/controleescolar/api/dto/SchoolDTO.java
package com.platformbuilders.controleescolar.api.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Min;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.validator.constraints.br.CNPJ;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class SchoolDTO {
    private Long id;

    @NotBlank(message = "O campo 'name' é obrigatório")
    private String name;

    @NotBlank(message = "O campo 'cpnj' é obrigatório")
    @CNPJ
    private String cnpj;

    @NotBlank(message = "O campo 'address' é obrigatório")
    private String address;

    @NotBlank(message = "O campo 'phone' é obrigatório")
    private String phone;

    @NotNull(message = "O campo 'studentsPerClassroom' é obrigatório")
    @Min(value = 1, message = "O campo 'studentsPerClassroom' deve ser maior que zero")
    private Integer studentsPerClassroom;
}

File: bff-controle-escolar/src/school/dto/school.dto.ts
import { Field, Int, ObjectType } from '@nestjs/graphql';
import { IsNotEmpty, IsString, IsInt, Min, IsOptional } from 'class-validator';

@ObjectType()
export class SchoolDTO {
  @Field(() => Int, { nullable: true })
  id?: number;

  @Field()
  @IsNotEmpty()
  @IsString()
  name: string;

  @Field()
  @IsNotEmpty()
  @IsString()
  cnpj: string;

  @Field()
  @IsNotEmpty()
  @IsString()
  address: string;

  @Field()
  @IsNotEmpty()
  @IsString()
  phone: string;

  @Field(() => Int)
  @IsInt()
  @Min(1)
  studentsPerClassroom: number;
}